scene/main/node.cpp

```diff
@@ -1906,7 +1906,7 @@ void Node::set_owner(Node *p_owner) {
 		check = check->data.parent;
 	}
 
-	ERR_FAIL_COND(!owner_valid);
+	ERR_FAIL_COND_MSG(!owner_valid, "Invalid owner. Owner must be a valid parent, grandparent, etc. ascending in the tree.");
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(!owner_valid, "Invalid owner. Owner must be an ancestor of the node in the tree.");
```
Would probably adjust the the wording in the docs to use ancestor as well and just elaborate that with the current explanation.

Reply:
How about changing the docs to:
"A node can have any other node as owner (as long as it is an ancestor, i.e. a valid parent, grandparent, etc. ascending in the tree)."

Reply:
Going a bit further:
```diff
diff --git a/doc/classes/Node.xml b/doc/classes/Node.xml
index 6bc167637e..36665e5543 100644
--- a/doc/classes/Node.xml
+++ b/doc/classes/Node.xml
@@ -858,8 +858,8 @@
 			[b]Note:[/b] Auto-generated names might include the [code]@[/code] character, which is reserved for unique names when using [method add_child]. When setting the name manually, any [code]@[/code] will be removed.
 		</member>
 		<member name="owner" type="Node" setter="set_owner" getter="get_owner">
-			The node owner. A node can have any other node as owner (as long as it is a valid parent, grandparent, etc. ascending in the tree). When saving a node (using [PackedScene]), all the nodes it owns will be saved with it. This allows for the creation of complex [SceneTree]s, with instancing and subinstancing.
-			[b]Note:[/b] If you want a child to be persisted to a [PackedScene], you must set [member owner] in addition to calling [method add_child]. This is typically relevant for [url=$DOCS_URL/tutorials/plugins/running_code_in_the_editor.html]tool scripts[/url] and [url=$DOCS_URL/tutorials/plugins/editor/index.html]editor plugins[/url]. If [method add_child] is called without setting [member owner], the newly added [Node] will not be visible in the scene tree, though it will be visible in the 2D/3D view.
+			The node owner. A node can have any ancestor node as owner (i.e. a parent, grandparent, etc. node ascending in the tree). This implies that [method add_child] should be called before setting the owner, so that this relationship of parenting exists. When saving a node (using [PackedScene]), all the nodes it owns will be saved with it. This allows for the creation of complex scene trees, with instancing and subinstancing.
+			[b]Note:[/b] If you want a child to be persisted to a [PackedScene], you must set [member owner] in addition to calling [method add_child]. This is typically relevant for [url=$DOCS_URL/tutorials/plugins/running_code_in_the_editor.html]tool scripts[/url] and [url=$DOCS_URL/tutorials/plugins/editor/index.html]editor plugins[/url]. If a new node is added to the tree without setting its owner as an ancestor in that tree,t will be visible in the 2D/3D view, but not in the scene tree (and not persisted when packing or saving).
 		</member>
 		<member name="process_mode" type="int" setter="set_process_mode" getter="get_process_mode" enum="Node.ProcessMode" default="0">
 			Can be used to pause or unpause the node, or make the node paused based on the [SceneTree], or make it inherit the process mode from its parent (default).
```

Reply:
Should I go ahead and add it to this or separately?

Reply:
I'd add this to the current commit, it's related.